### PR TITLE
Added UAI file format reader.

### PR DESCRIPTION
**Pull request by vivek425ster**

### CODE REVIEW

**Comment by ankurankan** [comment]
@vivek425ster Please let me know when this is ready for review.

**Comment by vivek425ster** [comment]
@ankurankan Please review this. 
I need to know how to get evidence in this.For now i got it using the same method as used in XMLBIF module. But in UAI evidence is expressed in entirely different file for UAI.
http://graphmod.ics.uci.edu/uai08/FileFormat. How to use that in get_model method?

**Comment by ankurankan** [comment]
@vivek425ster You don't need to parse the evidence files. We don't have any way of storing evidence in the model and we mostly pass the evidence as a parameter to methods where ever its required.

File: pgmpy/readwrite/UAI.py [hunk 1]
```diff
@@ -0,0 +1,256 @@
+from pyparsing import *
+from itertools import combinations
+from pgmpy.models import BayesianModel, MarkovModel
+from pgmpy.factors import TabularCPD, Factor
+import numpy as np
```
Comment (ankurankan) [inline pgmpy/readwrite/UAI.py]: @vivek425ster Import order PEP 8

File: pgmpy/readwrite/UAI.py [hunk 2]
```diff
@@ -0,0 +1,256 @@
...
+        self.edges = self.get_edges()
+        self.tables = self.get_tables()
+
+    def get_grammar(self):
+        """
+        Returns the grammar of the UAI file.
+        """
+        network_name = Word(alphas).setResultsName('network_name')
+        no_variables = Word(nums).setResultsName('no_variables')
+        grammar = network_name+no_variables
```
Comment (ankurankan) [inline pgmpy/readwrite/UAI.py]: @vivek425ster Space before and after +

File: pgmpy/readwrite/UAI.py [hunk 3]
```diff
@@ -0,0 +1,256 @@
...
+
+        Returns
+        -------
+        list: list of variables
+
+        Example
+        -------
+        >>> reader = UAIReader('TestUAI.uai')
+        >>> reader.get_variables()
+        ['var_0', 'var_1', 'var_2']
```
Comment (ankurankan) [inline pgmpy/readwrite/UAI.py:110]: @vivek425ster You should be mentioning this somewhere in the `get_model` method that the variable names would be in this pattern.

**Comment by ankurankan** [comment]
@vivek425ster There are few PEP8 issues (other than the one's I have commented on). Please fix those and remove the evidence part. Rest everything looks fine.

**Pull request merged**